### PR TITLE
Add new Authentication message for popcha

### PR DESCRIPTION
**Pull request by Jharaxus**
Adds partial Authentication message validation

### CODE REVIEW

**Review by K1li4nL — CHANGES_REQUESTED** [review]
All looking good ! Just a question and a tiny change request ;)

**Review by Ajkunas — CHANGES_REQUESTED** [review]
Good job :) 
I do not have a lot of comments on your PR. 
You should also take in account Kilian's comments.

File: be2-scala/src/main/scala/ch/epfl/pop/model/objects/Channel.scala [hunk 1]
```diff
@@ -61,6 +61,8 @@ object Channel {
   final val SOCIAL_MEDIA_CHIRPS_PREFIX: String = s"${SOCIAL_CHANNEL_PREFIX}chirps"
   final val REACTIONS_CHANNEL_PREFIX: String = s"${SOCIAL_CHANNEL_PREFIX}reactions"
 
+  final val POPCHA_AUTHENTICATION_LOCATION: String = s"${SOCIAL_CHANNEL_PREFIX}authentication"
```
[the remark this answers is not in the document]
Comment (Jharaxus) [reply be2-scala/src/main/scala/ch/epfl/pop/model/objects/Channel.scala:64]: You're right, I intended `ROOT_CHANNEL_PREFIX` instead!

File: be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidatorSuite.scala [hunk 2]
```diff
@@ -31,9 +31,6 @@ class ElectionValidatorSuite extends TestKit(ActorSystem("electionValidatorTestA
 
   final val DB_TEST_FOLDER: String = "databaseElectionTest"
 
-  val pubSubMediatorRef: ActorRef = system.actorOf(PubSubMediator.props, "PubSubMediator")
-  val dbActorRef: AskableActorRef = system.actorOf(Props(DbActor(pubSubMediatorRef, MessageRegistry(), InMemoryStorage())), "DbActor")
```
[the remark this answers is not in the document]
Comment (Jharaxus) [reply be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidatorSuite.scala:35]: Yes, I came across that code while writing my own tests which IntelliJ told me was unused, so I removed it while I was at it as I was confused trying to find out what it was needed for, when actually it wasn't used at all (probably the leftovers of a copy/paste from other tests that need it)

**Review by K1li4nL — DISMISSED** [review]
LGTM !

File: be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/validators/PopchaValidatorSuite.scala [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+import org.scalatest.funsuite.{AnyFunSuiteLike => FunSuiteLike}
+import org.scalatest.matchers.should.Matchers
+import util.examples.JsonRpcRequestExample._
+import util.examples.MessageExample
+
+import java.io.File
+import java.util.concurrent.TimeUnit
+import scala.reflect.io.Directory
+
+class PopchaValidatorSuite extends TestKit(ActorSystem("electionValidatorTestActorSystem")) with FunSuiteLike
```
Comment (Ajkunas) [inline be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/validators/PopchaValidatorSuite.scala:21]:
```suggestion
class PopchaValidatorSuite extends TestKit(ActorSystem("popChaValidatorTestActorSystem")) with FunSuiteLike
```

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/PopchaValidator.scala [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  }
+
+  private def checkResponseMode(rpcMessage: JsonRpcRequest, responseMode: String, error: PipelineError): GraphMessage = {
+    if (responseMode == "query" || responseMode == "fragment")
+      Right(rpcMessage)
+    else
+      Left(error)
+  }
+
+  private def checkChannelName(rpcMessage: JsonRpcMessage, channel: Channel, laodId: Hash, error: PipelineError): GraphMessage = {
```
Comment (Ajkunas) [inline be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/PopchaValidator.scala:61]: Now that I think about it, why don't you use `checkChannelType` ? If I am not mistaken, that function checks the existence of the Channel, thus the laoId.

**Review by Ajkunas — CHANGES_REQUESTED** [review]
Last little comments :) Do not hesitate to discuss my comments.

File: be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/validators/PopchaValidatorSuite.scala (hunk 3 above)
Comment (Jharaxus) [reply be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/validators/PopchaValidatorSuite.scala:21]: Indeed I forgot to change that !

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/PopchaValidator.scala (hunk 4 above)
Comment (Jharaxus) [reply be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/PopchaValidator.scala]: Indeed!

**Review by Ajkunas — APPROVED** [review]
LGTM

**Pull request merged**